code/go/0chain.net/blobbercore/challenge/protocol.go

```diff
@@ -81,6 +84,10 @@ func (cr *ChallengeEntity) LoadValidationTickets(ctx context.Context) error {
 		return err
 	}
 
+	// Lock allocation changes from happening in handler.CommitWrite function
+	allocMu := lock.GetMutex(allocationObj.TableName(), allocationObj.ID)
+	allocMu.Lock()
+
```

Comment:
defer allocMu.Unlock()

Reply:
@sum-it08 deferring is not good choice here. Unlock should be called as early as possible because it would be locking `handler.CommitWrite` which makes allocation changes.

Deferring will not unlock until it reaches `return` statement.